test/framework/workload_cluster.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// podListCondition is a type that operates a condition on a Pod
+type podListCondition func(p *corev1.PodList) error
+
+// WaitForPodListConditionInput is the input args for WaitForPodListCondition
+type WaitForPodListConditionInput struct {
+	Lister      Lister
+	ListOptions *client.ListOptions
+	Condition   podListCondition
+}
+
+// WaitForPodListCondition waits for the specified condition to be true for all
+// pods returned from the list filter.
+func WaitForPodListCondition(ctx context.Context, input WaitForPodListConditionInput, intervals ...interface{}) {
```

Comment:
We can use this to also test the CoreDNS images having the correct image tags after a coredns upgrade.

Reply:
@gab-satchi If you'd like you can also just do the image check on the coredns `deployment.spec.containers[0].Image` similar to how @sedefsavas did for kube-proxy.
I had to do a pod list because etcd is deployed as static pods.